developer/docs/README.md

```diff
@@ -0,0 +1,30 @@
+# Documentation for Developers
```

Comment:
Not necessarily a change in this PR, but I would like to improve this content in a future PR to explain we have the devenv env and local env and ckcp env for developers to test Pipeline Service.